### PR TITLE
chore: test axe.runPartial deepEqual axe.run

### PR DESCRIPTION
**Pull request by WilcoFiers**

### CODE REVIEW

File: packages/playwright/tests/axe-playwright.spec.ts [hunk 1]
```diff
@@ -17,16 +17,20 @@ describe('@axe-core/playwright', () => {
   const axeSource = fs.readFileSync(axePath, 'utf8');
   let browser: playwright.ChromiumBrowser;
   const axeTestFixtures = path.resolve(__dirname, 'fixtures');
+  const externalPath = path.resolve(axeTestFixtures, 'external');
   const axeLegacySource = fs.readFileSync(
-    path.resolve(axeTestFixtures, 'external', 'axe-core@legacy.js'),
+    `${externalPath}/axe-core@legacy.js`,
```
Comment (straker) [inline packages/playwright/tests/axe-playwright.spec.ts:22]: This path structure could fail on windows

```suggestion
    path.join(externalPath, 'axe-core@legacy.js'),
```

File: packages/playwright/tests/axe-playwright.spec.ts [hunk 2]
```diff
@@ -17,16 +17,20 @@ describe('@axe-core/playwright', () => {
...
     'utf-8'
   );
-  const axeCrashPath = path.resolve(
-    axeTestFixtures,
-    'external',
-    'axe-crasher.js'
+  const axeCrasherSource = fs.readFileSync(
+    `${externalPath}/axe-crasher.js`,
```
Comment (straker) [inline packages/playwright/tests/axe-playwright.spec.ts:26]:
```suggestion
    path.join(externalPath, 'axe-crasher.js')
```

File: packages/playwright/tests/axe-playwright.spec.ts [hunk 3]
```diff
@@ -17,16 +17,20 @@ describe('@axe-core/playwright', () => {
...
+    'utf8'
   );
-  const axeCrasherSource = fs.readFileSync(axeCrashPath, 'utf8');
+  const axeForceLegacy = fs.readFileSync(
+    `${externalPath}/axe-force-legacy.js`,
```
Comment (straker) [inline packages/playwright/tests/axe-playwright.spec.ts:30]:
```suggestion
    path.join(externalPath, 'axe-force-legacy.js'),
```

File: packages/puppeteer/test/axePuppeteer.test.ts [hunk 4]
```diff
@@ -32,15 +32,20 @@ describe('AxePuppeteer', function () {
...
   let axeCrasherSource: string;
+  let axeForceLegacy: string;
 
   before(async () => {
     const axePath = require.resolve('axe-core');
     axeSource = fs.readFileSync(axePath, 'utf8');
-    const axeCrashPath = path.resolve(
-      __dirname,
-      './fixtures/external/axe-crasher.js'
+    const externalPath = path.resolve(__dirname, './fixtures/external/');
```
Comment (straker) [inline packages/puppeteer/test/axePuppeteer.test.ts:40]:
```suggestion
    const externalPath = path.join(__dirname, 'fixtures', 'external');
```

File: packages/puppeteer/test/axePuppeteer.test.ts [hunk 5]
```diff
@@ -32,15 +32,20 @@ describe('AxePuppeteer', function () {
...
+    axeCrasherSource = fs.readFileSync(
+      `${externalPath}/axe-crasher.js`,
```
Comment (straker) [inline packages/puppeteer/test/axePuppeteer.test.ts:42]:
```suggestion
      path.join(externalPath, 'axe-crasher.js'),
```

File: packages/puppeteer/test/axePuppeteer.test.ts [hunk 6]
```diff
@@ -32,15 +32,20 @@ describe('AxePuppeteer', function () {
...
+      'utf8'
+    );
+    axeForceLegacy = fs.readFileSync(
+      `${externalPath}/axe-force-legacy.js`,
```
Comment (straker) [inline packages/puppeteer/test/axePuppeteer.test.ts:46]:
```suggestion
      path.join(externalPath, 'axe-force-legacy.js'),
```

File: packages/webdriverio/src/test.ts [hunk 7]
```diff
@@ -74,8 +74,16 @@ describe('@axe-core/webdriverio', () => {
       path.resolve(axeTestFixtures, 'axe-core@legacy.js'),
       'utf-8'
     );
-    const axeCrashPath = path.resolve(axeTestFixtures, 'axe-crasher.js');
-    const axeCrasherSource = fs.readFileSync(axeCrashPath, 'utf8');
+    const axeCrasherSource = fs.readFileSync(
+      `${axeTestFixtures}/axe-crasher.js`,
```
Comment (straker) [inline packages/webdriverio/src/test.ts:78]:
```suggestion
      path.join(axeTestFixtures, 'axe-crasher.js'),
```

File: packages/webdriverio/src/test.ts [hunk 8]
```diff
@@ -74,8 +74,16 @@ describe('@axe-core/webdriverio', () => {
...
+      'utf8'
+    );
+    const axeForceLegacy = fs.readFileSync(
+      `${axeTestFixtures}/axe-force-legacy.js`,
```
Comment (straker) [inline packages/webdriverio/src/test.ts:82]:
```suggestion
      path.join(axeTestFixtures, 'axe-force-legacy.js'),
```

**Review by straker — DISMISSED** [review]

File: packages/playwright/tests/axe-playwright.spec.ts [hunk 9]
```diff
@@ -353,6 +371,20 @@ describe('@axe-core/playwright', () => {
...
+      const axeSourceNoPartial =
+        axeSource +
+        `;delete window.axe.runPartial; delete window.axe.finishRun;`;
+      const legacyResults = await new AxeBuilder({
+        page,
+        axeSource: axeSourceNoPartial
+      }).analyze();
+      normalResults.timestamp = legacyResults.timestamp;
+      assert.deepEqual(normalResults, legacyResults);
+    });
```
Comment (straker) [inline packages/playwright/tests/axe-playwright.spec.ts:387]: Don't need to run the test twice

```suggestion
```

File: packages/puppeteer/test/axePuppeteer.test.ts [hunk 10]
```diff
@@ -692,6 +711,18 @@ describe('AxePuppeteer', function () {
...
+      await page.goto(`${addr}/nested-iframes.html`);
+      const normalResults = await new AxePuppeteer(page, axeSource).analyze();
+      const legacyResults = await new AxePuppeteer(
+        page,
+        axeSource +
+          `;delete window.axe.runPartial; delete window.axe.finishRun;`
+      ).analyze();
+      normalResults.timestamp = legacyResults.timestamp;
+      assert.deepEqual(normalResults, legacyResults);
+    });
```
Comment (straker) [inline packages/puppeteer/test/axePuppeteer.test.ts:725]: Don't need to run the test twice

```suggestion
```

File: packages/webdriverio/src/test.ts [hunk 11]
```diff
@@ -563,6 +588,27 @@ describe('@axe-core/webdriverio', () => {
...
+            source +
+            `;delete window.axe.runPartial; delete window.axe.finishRun;`;
+          const legacyResults = await new AxeBuilder({
+            client,
+            axeSource: axeSourceNoPartial
+          }).analyze();
+
+          normalResults.timestamp = legacyResults.timestamp;
+          assert.deepEqual(normalResults, legacyResults);
+        });
```
Comment (straker) [inline packages/webdriverio/src/test.ts:611]: Don't need to run the test twice

```suggestion
```

**Review by straker — CHANGES_REQUESTED** [review]

File: packages/puppeteer/test/axePuppeteer.test.ts (hunk 4 above)
Comment (straker) [reply packages/puppeteer/test/axePuppeteer.test.ts:40]: This is still there

**Review by straker — DISMISSED** [review]

**Comment by straker** [comment]
Looks like you might need to update from latest again to get chromedriver changes https://github.com/dequelabs/axe-core-npm/pull/355

**Review by michael-siek — APPROVED** [review]

**Pull request merged**